Challenge-4/Bot_2/src/agent.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  HandleTransaction,
+  TransactionEvent,
+  ethers,
+  Finding,
+  getEthersProvider,
+} from "forta-agent";
+import {
+  BORROW_RATE,
+  SUPPLY,
+  SUPPLY_RATE,
+  USDC_TOKEN_ETH,
+  UTILIZATION,
+  WITHDRAW,
+} from "./constants";
+import Helper from "./helper";
+import CONFIG from "./agent.config";
+import { NetworkData } from "./types";
+import { NetworkManager } from "forta-agent-tools";
+import {
+  supplyFinding,
+  borrowFinding,
+  alertSupplyFinding,
+  alertBorrowFinding,
+} from "./findings";
+
+let configuratorProxy: string | undefined;
+let tokenAddress: string;
+let helper: Helper;
+const networkManager = new NetworkManager<NetworkData>(CONFIG);
+export function provideInitialize(provider: ethers.providers.Provider) {
+  return async function initialize() {
+    await networkManager.init(provider);
+    helper = new Helper(provider);
+  };
+}
+
+export function provideHandleTransaction(
+): HandleTransaction {
+  return async function HandleTransaction(tx: TransactionEvent) {
+    const finding: Finding[] = [];
+
+    tokenAddress = networkManager.get("usdc");
+
+    const withdraw = tx.filterFunction([WITHDRAW], tokenAddress);
+
+    const supply = tx.filterFunction([SUPPLY], tokenAddress);
+
+    configuratorProxy = networkManager.get("configurationProxy");
+
+    // get configuration values from the configurator contract for USDC token
+    const configuration = await helper.getProtocolConfiguration(
+      USDC_TOKEN_ETH,
+      configuratorProxy,
+      tx.blockNumber,
+    );
+
+    // get utilization for cUSDC token
+    const utilization = await helper.getUtilization(
+      tokenAddress,
+      UTILIZATION,
+      tx.blockNumber,
+    );
+
+    // get annual percentage rate for supply
+    const supplyAPR = await helper.getSupplyAPR(
+      tokenAddress,
+      SUPPLY_RATE,
+      utilization,
+      tx.blockNumber,
+    );
+
+    // get annual percentage rate for borrow
+    const borrowAPR = await helper.getBorrowAPR(
+      tokenAddress,
+      BORROW_RATE,
+      utilization,
+      tx.blockNumber,
+    );
```

Comment:
I feel like you could implement a function in the `Helper` class that is called by `agent.ts`, and accepts the arguments needed by all of these function calls. Then _that_ function calls all of the needed functions in `Helper`, and returns the four values here. That would clean up the bot here. So `agent.ts` calls this "main" function in `helper.ts`, and that "main" function calls `getProtocolConfiguration()`, `getUtilization()`, etc.

Also, just using "main" as an example name.